src/coinbase/address/wallet_address.ts

```diff
@@ -0,0 +1,364 @@
+import { Decimal } from "decimal.js";
+import { ethers } from "ethers";
+import { Address } from "../address";
+import { Address as AddressModel } from "../../client";
+import { Asset } from "../asset";
+import { Balance } from "../balance";
+import { BalanceMap } from "../balance_map";
+import { Coinbase } from "../coinbase";
+import { ArgumentError, InternalError } from "../errors";
+import { FaucetTransaction } from "../faucet_transaction";
+import { Trade } from "../trade";
+import { Transfer } from "../transfer";
+import { Amount, Destination, TransferStatus } from "../types";
+import { delay } from "../utils";
+import { Wallet as WalletClass } from "../wallet";
+
+/**
+ * A representation of a blockchain address, which is a wallet-controlled account on a network.
+ */
+export class WalletAddress extends Address {
+  private _model: AddressModel;
+  private _key?: ethers.SigningKey;
```

Comment:
nit: we don't use _ prefix since private is present in the line.

```suggestion
  private model: AddressModel;
  private key?: ethers.SigningKey;
```